src/EnergyPlus/WeatherManager.cc

```diff
@@ -3657,16 +3683,44 @@ namespace WeatherManager {
                     TomorrowWindSpeed(TS, Hour) = LastHrWindSpeed * WtPrevHour + Wthr.WindSpeed(Hour) * WtNow;
                     TomorrowWindDir(TS, Hour) = interpolateWindDirection(LastHrWindDir, Wthr.WindDir(Hour), WtNow);
                     TomorrowHorizIRSky(TS, Hour) = LastHrHorizIRSky * WtPrevHour + Wthr.HorizIRSky(Hour) * WtNow;
-                    if (Environment(Envrn).SkyTempModel > 3 || Environment(Envrn).SkyTempModel == 0) {
-                        TomorrowSkyTemp(TS, Hour) = LastHrSkyTemp * WtPrevHour + Wthr.SkyTemp(Hour) * WtNow;
-                    }
                     TomorrowDifSolarRad(TS, Hour) =
                         LastHrDifSolarRad * WgtPrevHour + Wthr.DifSolarRad(Hour) * WgtHourNow + NextHrDifSolarRad * WgtNextHour;
                     TomorrowBeamSolarRad(TS, Hour) =
                         LastHrBeamSolarRad * WgtPrevHour + Wthr.BeamSolarRad(Hour) * WgtHourNow + NextHrBeamSolarRad * WgtNextHour;
 
                     TomorrowLiquidPrecip(TS, Hour) = LastHrLiquidPrecip * WtPrevHour + Wthr.LiquidPrecip(Hour) * WtNow;
                     TomorrowLiquidPrecip(TS, Hour) /= double(NumOfTimeStepInHour);
+                    TomorrowTotalSkyCover(TS, Hour) = LastHrTotalSkyCover * WtPrevHour + Wthr.TotalSkyCover(Hour) * WtNow;
+                    TomorrowOpaqueSkyCover(TS, Hour) = LastHrOpaqueSkyCover * WtPrevHour + Wthr.OpaqueSkyCover(Hour) * WtNow;
+
+                    Real64 InterpOpaqueSkyCover = TomorrowOpaqueSkyCover(TS, Hour); 
+                    Real64 InterpDryBulb = TomorrowOutDryBulbTemp(TS, Hour); 
+                    Real64 InterpDewPoint = TomorrowOutDewPointTemp(TS, Hour); 
+                    Real64 InterpRelHumDeci = TomorrowOutRelHum(TS, Hour) * 0.01; 
+                    Real64 InterpIRHoriz = TomorrowHorizIRSky(TS, Hour); 
+
+                    // Sky emissivity now takes interpolated timestep inputs rather than interpolated calcation esky results
```

Comment:
This section duplicates calculations done earlier in lines 3516:3535. Are those line still necessary?

Reply:
Lines 3516:3535 is exactly the original way of calculating hourly values of HorizIRSky and SkyTemp and then interpolate here. Theoretically, yes, with the new method implemented, the lines from 3516 and 3535 can be removed.

Reply:
Lines 3516:3535 is exactly the original way of calculating hourly values of HorizIRSky and SkyTemp and then interpolate here. Theoretically, yes, with the new method implemented, the lines from 3516 and 3535 can be removed.

Reply:
@mjwitte The code has been revised based on the few discussions above. The lines 3516:3535 is kept though, since it preserved a section of calculating the hourly values and part of it (the `HorizIR` involves in an interpolation in one of the if branches.

Reply:
Then here, check for `TomorrowHorzIRSky(1, Hour) >= 9999` ?

Reply:
@mjwitte Agree, except that I think here we should check `TomorrowHorizIRSky(TS, Hour) >= 9999` instead, since `TomorrowHorizIRSky(1, Hour)` would have been override when doing `(<2,3,4...>, Hour)`. This would be well with your suggestion on doing the direct copy for weather on `TomorrowHorizIRSky(CurTimeStep, Hour) = IRHoriz;` on Line 3516.

Reply:
@jcyuan2020 I see your point, but I don't think you'll find all of the TS values are filled in. The loop on `CurTimeStep <= NumIntervalsPerHour` is based on the incoming epw data timestep. For hourly weather files, `NumIntervalsPerHour` is 1.

Reply:
So, perhaps a local bool at the top of the hour loop to check if HorizIRSky was missing for this hour?

Reply:
@mjwitte I see. Would a bool flag have the same effect of Line 3516  `HrIRHoriz_Weather(Hour) = IRHoriz;` and then later check on Lines 3703 and 3714 Hr  `HrIRHoriz_Weather(Hour) >= 9999.0`. This was the mechanism added to the most recent revision (about 10 days ago) to address how to check if hourly IRHoriz data is missing for the hour.

Reply:
Actually, the bool isn't necessary. The TS=1 values are saved in `Wthr`, so just test here on `Wthr.HorizIRSky(Hour)`.

Reply:
@mjwitte I see. This will then invoke again the requirement for keeping Lines 3516-3535. Also this is required due to the `if` condition in Line 3602 `NumOfTimeStepInHour > 1`--if Lines 3516-3535 are removed, then the situation of using hourly timestep (when `NumOfTimeStepInHour ==1`), the HorizIR and SkyTemp would not be filled and updated correctly since this whole block of interpolation from Lines 3602-3745 would not run when `NumOfTimeStepInHour` is equal to 1.

Reply:
Good point. OK, so I guess there's no way around having both sections. 
If it makes sense to move this to a small function to avoid code duplication, please do that. 
Should the code at 3156 include an `if (NumIntervalsPerHour > 1 ||  NumOfTimeStepInHour == 1)`?
I'll stop commenting here and let you work out a final solution (which may look much like what you had before, sorry).

Reply:
@mjwitte Thank you for the comments and they helped me to understand the problems much better. The code has been revised per the recent discussions above.